plugins/hydra_ax_sweeper/hydra_plugins/hydra_ax_sweeper/_core.py

```diff
@@ -76,14 +84,20 @@ def get_one_batch_of_trials(
 
     batch_of_trials = []
     for _ in range(max_parallelism_setting):
-        parameters, trial_index = ax_client.get_next_trial()
-        batch_of_trials.append(
-            Trial(
-                overrides=map_params_to_arg_list(params=parameters),
-                trial_index=trial_index,
+        try:
+            parameters, trial_index = ax_client.get_next_trial()
+        except SearchSpaceExhausted as exception:
+            is_search_space_exhausted = True
+            exception_msg = str(exception)
+            break
+        else:
+            batch_of_trials.append(
+                Trial(
+                    overrides=map_params_to_arg_list(params=parameters),
+                    trial_index=trial_index,
+                )
```

Comment:
I never liked try, except else.
why not something like this:
```suggestion
        try:
            parameters, trial_index = ax_client.get_next_trial()
            batch_of_trials.append(
                Trial(
                    overrides=map_params_to_arg_list(params=parameters),
                    trial_index=trial_index,
                )            
        except SearchSpaceExhausted as exception:
            is_search_space_exhausted = True
            exception_msg = str(exception)
            break
```
?